editor.planx.uk/src/components/Feedback/index.test.tsx

```diff
@@ -0,0 +1,369 @@
+import { Breadcrumbs } from "@opensystemslab/planx-core/types";
+import { act, waitFor } from "@testing-library/react";
+import {
+  getInternalFeedbackMetadata,
+  insertFeedbackMutation,
+} from "lib/feedback";
+import { vanillaStore } from "pages/FlowEditor/lib/store";
+import React from "react";
+import { axe, setup } from "testUtils";
+
+import Feedback from "./index";
+
+const { setState } = vanillaStore;
+
+const mockedBreadcrumbs: Breadcrumbs = {
+  LU5xin8PHs: {
+    answers: ["xuFhs3Hqr9"],
+    auto: false,
+  },
+};
+
+jest.mock("lib/feedback", () => ({
+  getInternalFeedbackMetadata: jest.fn(),
+  insertFeedbackMutation: jest.fn(),
+}));
+
+const scrollIntoViewMock = jest.fn();
+window.Element.prototype.scrollIntoView = scrollIntoViewMock;
+
+describe("Feedback component triage journey", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test("Initial render shows the FeedbackPhaseBannerView but doesn't scroll into view", () => {
+    const { getByText } = setup(<Feedback />);
+
+    expect(getByText("PUBLIC BETA")).toBeInTheDocument();
+    expect(getByText("Report an issue with this page")).toBeInTheDocument();
+    expect(getByText("feedback")).toBeInTheDocument();
+    expect(scrollIntoViewMock).not.toBeCalled();
+  });
+
+  test("Selecting 'feedback' scrolls triage into view", async () => {
+    const { getByText, getByRole, user } = setup(<Feedback />);
+
+    await user.click(getByText("feedback"));
+
+    expect(scrollIntoViewMock).toBeCalledTimes(1);
+
+    await waitFor(() => {
+      expect(getByRole("button", { name: "Issue" })).toBeInTheDocument();
+      expect(getByRole("button", { name: "Idea" })).toBeInTheDocument();
+      expect(getByRole("button", { name: "Comment" })).toBeInTheDocument();
+    });
```

Comment:
I'm really glad you brought this up Jess as it spurred me on to get a more exact answer. 

I tried this out and as per yesterday the test still passes without the `await waitFor()`.

Although when I was writing stuff today I fell back in the habit of wrapping things in `await waitFor()` as it's a pattern I'm very accustomed to and I fell afoul of async behaviour.

I had another looks and I believe it's best practice to use the `await waitFor()` pattern for the result of anything which is asynchronous. 

When I'm looking for the Feedback view to have changes with these `expect` statement it's as a result of `setState` which I believe is asynchronous as well as also the `handleSubmit` which is definitely asynchronous. 

I think the test are passing because the setup in the test means there shouldn't be delays but I _think_ there is a risk it could catch us out?

Although I might be missing something or be working off of out of date info so more than happy to change if that's the case.

Reply:
Thanks for explanation and following up about this one - happy to defer to your judgement - I think I've just been used to seeing `await waitFor()` pattern used around more explicit mock API calls or `handleSubmit` or similar rather than checking for existence of content.